Feature/090 filter with api

Joppe-VH opened the pull request:
filter is working except for the price selection. Tho that does work on the frontend, the backend doesn't support it yet.

I moved the vast majority of the state to the url. After a lot of trial and error this was by far the best option.
It also means that people can copy and share the url with eachother.

the shop page now also has a pagination param in the url because of this.

sorting and ordering works too.

amnabbouti reviewed (changes requested):
Noticed a couple of things:

- in `pokemartApi.ts`  at `getItems` you are sending empty arrays as empty strings but numeric parameters aren’t validated properly that will cause an issue! a better approach would be to filter out invalid values before sending them to the API.

- Clicking on an article doesn’t fetch the details, could be a routing issue or possibly something off in ProductService (not sure) might be worth checking if the product ID is being handled correctly.

Joppe-VH commented:
@amnabbouti think I adressed most of the issues you mentioned

I also made sure that the page resets to 1 when you change the filter settings
an additional thing I added (because it was a bit related) is that I finished up the styling of the sort and order by fields.

Finally the detail page link, that does work, but I think the hosted backend is somehow not on the latest version.
It is putting undefined in the link, because it can't find the slug property on an item
this is very weird because I have checked all our mongoose databases and they all, dev, prod, and my local one have the same updated data with the slug property on the items. Also I'm not sure how to add sensitive ENV variables, since I can't simply put them in the docker file like how you did the PORT env variable.

I've got the render one back up and running for both dev and main. I've switched us back to those two for now since they seem to work.

Joppe-VH commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Integrate shop grid and filter with redux API.](https://trello.com/c/FHAOZCgW/90-integrate-shop-grid-and-filter-with-redux-api)
